app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+
+  REGEXP_EMAIL = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+  
+  enum role: [:customer, :admin]
+  has_secure_password
+
+  validates :name, presence: true
+  validates :email, uniqueness: true, format: {
```

Comment:
add allow blank here, as uniqueness always fire one extra query. We can save it in case no email ID was present